src/NuGet.Core/NuGet.Configuration/Settings/Items/AuthorItem.cs

```diff
@@ -27,7 +27,7 @@ internal AuthorItem(XElement element, SettingsFile origin)
 
         public override SettingBase Clone()
         {
-            var newItem = new AuthorItem(Name, Certificates.Select(c => c.Clone() as CertificateItem).ToArray());
+            var newItem = new AuthorItem(Name, Certificates.Select(c => (CertificateItem)c.Clone()).ToArray());
```

Comment:
I doubt it matters in this code path but before the array of certificates would be a clone and a safe cast to `CertificateItem` so the array _could_ have `null`s?  Now it will throw?  Again, probably doesn't matter, just wanted to point it out in case it does.

Reply:
It's not possible, because `Certificates` has type `IList<CertificateItem>`. The implementation of `CertificateItem.Clone()` returns a `new CertificateItem(...)`, but the `Clone` method is defined in `SettingsBase`, so `Clone()`'s return type is `SettingBase`.  So, IMO it's an example of incorrect usage of the `as` keyword.

Personally, I think every class should use the pattern

```c#
public SettingsBase SettingsBase.Clone() => Clone();

public new CertificateItem Clone() { /* implementation */ }
```

This way callers won't need to cast, and people reading the code would be less confused.  But I didn't want to spend a lot of time changing the assembly, just trying to annotate "as quickly as possible".